core/src/main/java/org/elasticsearch/index/query/TermsQueryBuilder.java

```diff
@@ -30,96 +54,209 @@
 
     public static final String NAME = "terms";
 
-    static final TermsQueryBuilder PROTOTYPE = new TermsQueryBuilder(null, (Object) null);
+    static final TermsQueryBuilder PROTOTYPE = new TermsQueryBuilder(null);
 
-    private final String name;
-
-    private final Object values;
+    public static final boolean DEFAULT_DISABLE_COORD = false;
 
+    private final String fieldName;
+    private final List<Object> values = new ArrayList<>();
     private String minimumShouldMatch;
+    private boolean disableCoord = DEFAULT_DISABLE_COORD;
+    private TermsLookupBuilder termsLookup;
+
+    public static final class TermsLookupBuilder implements Writeable<TermsLookupBuilder>, ToXContent {
+        static final TermsLookupBuilder PROTOTYPE = new TermsLookupBuilder();
+
+        private String index;
+        private String type;
+        private String id;
+        private String path;
+        private String routing;
+
+        public TermsLookupBuilder() {
```

Comment:
No, you are right, I didn't realize the need for api users before going through the whole changes.